README.md
# Rise Vision Developer Hub

## Introduction

Rise vision developer hub is an application built with [AngularJs](https://angularjs.org/) its plugin and our [Common-header](https://github.com/Rise-Vision/common-header) to provide a space where developer can manage their apps, have access to developer documentation and play with our developer tools.

Project Name works in conjunction with [Rise Vision](http://www.risevision.com), the [digital signage management application](http://rva.risevision.com/) that runs on [Google Cloud](https://cloud.google.com).

At this time Chrome is the only browser that this project and Rise Vision supports.

## Built With
- *[NPM](https://www.npmjs.org/)*
- *[AngularJs](https://angularjs.org/)*
- *[Bower](http://bower.io/)*
- *[Gulp.js](http://gulpjs.com/)*
- *[Common-Style](http://rise-vision.github.io/style-guide/)*
- *[common-header](https://github.com/Rise-Vision/common-header)*
- *[JQuery](http://jquery.com/)*

## Development

### Local Development Environment Setup and Installation

* install npm

* clone the repo using Git to your local:
```bash
git clone https://github.com/Rise-Vision/developer-hub.git
```

* install all javascript libs such as gulp
```bash
npm install
```

* download components with bower
```bash
gulp bower-clean-install
```

### Run Local

* under the root directory run gulp default task which will build and watch directories for changes 
so building and reloading the browser after a change
```bash
gulp
```

* If browser hasn't opened, you can open it on http://localhost:8000

### Dependencies


### Testing

## Submitting Issues
If you encounter problems or find defects we really want to hear about them. If you could take the time to add them as issues to this Repository it would be most appreciated. When reporting issues please use the following format where applicable:

**Reproduction Steps**

1. did this
2. then that
3. followed by this (screenshots / video captures always help)

**Expected Results**

What you expected to happen.

**Actual Results**

What actually happened. (screenshots / video captures always help)

## Contributing
All contributions are greatly appreciated and welcome! If you would first like to sound out your contribution ideas please post your thoughts to our [community](http://community.risevision.com), otherwise submit a pull request and we will do our best to incorporate it

### Languages

In order to support languages i18n needs to be added to this repository.  Please refer to our Suggested Contributions.

### Suggested Contributions
- We could use i18n Language Support

## Resources
If you have any questions or problems please don't hesitate to join our lively and responsive community at http://community.risevision.com.

If you are looking for user documentation on Rise Vision please see http://www.risevision.com/help/users/

If you would like more information on developing applications for Rise Vision please visit http://www.risevision.com/help/developers/.

**Facilitator**

[Rodrigo Serviuc Pavezi](https://github.com/rodrigopavezi "Rodrigo Serviuc Pavezi")


CHANGELOG.md
# Changelog

## [1.0-beta] (https://github.com/Rise-Vision/developer-hub/tree/v1.0-beta)
#### 28/11/14


\* *This changelog was generated by [github_changelog_generator](https://github.com/skywinder/Github-Changelog-Generator)*

gulpfile.js
/**
 * This script is based on https://github.com/shakyShane/jekyll-gulp-sass-browser-sync
 * Created by rodrigopavezi on 10/6/14.
 */
var env = process.env.NODE_ENV || "dev"

var gulp        = require('gulp');
var gutil       = require('gulp-util');
var browserSync = require('browser-sync');
var prefix      = require('gulp-autoprefixer');
var bower       = require('bower');
var del         = require('delete');
var deploy      = require('gulp-gh-pages');
var argv        = require('minimist')(process.argv.slice(2));
var rename      = require("gulp-rename");
var factory     = require("widget-tester").gulpTaskFactory;
var runSequence = require("run-sequence");


//------------------------- Browser Sync --------------------------------

/**
 * Wait for jekyll-build, then launch the Server
 */
gulp.task('browser-sync', function() {
    browserSync({
        server: {
            baseDir: 'web'
        },
        port: 8000
    });
});

gulp.task('browser-sync-reload', function() {
    browserSync.reload();
});

//------------------------- Bower --------------------------------

/**
 * Install bower dependencies
 */
gulp.task('bower-install', ['bower-clean-cache', 'bower-rm'], function(){
    return bower.commands.install([], {save: true}, {});
});

/**
 * Clean bower cache
 */
gulp.task('bower-clean-cache', function(){
    return bower.commands.cache.clean([], {}, {});
});

/**
 *  Remove all bower dependencies
 */
gulp.task('bower-rm', function(){
    return del.sync('web/components');
});

//------------------------- Watch --------------------------------
/**
 * Watch scss files for changes & recompile
 * Watch html/md files, run jekyll & reload BrowserSync
 */
gulp.task('watch', function () {
    gulp.watch(['web/**/*'], ['browser-sync-reload']);
});

//-------------------------- Test ----------------------------------
gulp.task("server", factory.testServer());
gulp.task("server-close", factory.testServerClose());
gulp.task("test:webdrive_update", factory.webdriveUpdate());
gulp.task("test:e2e:core", ["test:webdrive_update"], factory.testE2EAngular({
    browser: "chrome"
}));
gulp.task("test:e2e", function (cb) {
    runSequence("server", "test:e2e:core", "server-close", cb);
});


//------------------------- Deployment --------------------------------
var options = {
            prod: {
                remoteUrl: "git@github.com:Rise-Vision/dev-hub-prod.git"
            },
            stage: {
                remoteUrl: "git@github.com:Rise-Vision/dev-hub-stage.git"
            }
        };

/**
 *  Deploy to gh-pages
 */
gulp.task("deploy", function () {

    // Remove temp folder created by gulp-gh-pages
    if (argv.clean) {
        var os = require('os');
        var path = require('path');
        var repoPath = path.join(os.tmpdir(), 'tmpRepo');
        gutil.log('Delete ' + gutil.colors.magenta(repoPath));
        del.sync(repoPath, {force: true});
    }

    return gulp.src("./web/**/*")
        .pipe(deploy(options[env]));
});


/**
 * Copy and rename CNAME file depending on the target environment
 */
gulp.task("cname", function() {
    gulp.src("./cname-config/CNAME-"+env)
    .pipe(rename("CNAME"))
    .pipe(gulp.dest("./web"));
});


/**
 * Do a bower clean install
 */
gulp.task('bower-clean-install', ['bower-rm', 'bower-clean-cache','bower-install']);

/**
 * Default task, running just `gulp` will compile the sass,
 * compile the jekyll site, launch BrowserSync & watch files.
 */
gulp.task('default', ['browser-sync', 'watch']);




